ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/ActionManager.java

```diff
@@ -39,9 +39,26 @@ public void remove(Uid id)
 		_allActions.remove(id);
 	}
 
-    public int getNumberOfInflightTransactions()
+	/**
+	 * <p>
+	 *     This method works out the number of in-flight transactions currently in
+	 *     the system.
+	 * </p>
+	 * <p>
+	 *     Note: The definition of in-flight (AKA active) transactions, can be found
```

Comment:
I think we could change this to an Arjuna definition.

Note: The definition of in-flight transactions are those BasicActions that have status() of ActionStatus.RUNNING

BTW I wonder if we should be counting CREATED transactions too in the filter? In which case if we change the filter to be that it would be: "The definition of in-flight transactions are those BasicActions that have created and not yet begun, or a status() of ActionStatus.RUNNING"

Reply:
That's an unusual definition since it would include transactions that haven't started yet. If we want include those then we would need to add a new stat for it.

Reply:
WDYT?

Note: The definition of in-flight (AKA active) transactions, can be found in the OMG OTS specification: [active is defined as] the state of a transaction when processing is in progress and completion of the transaction has not yet commenced. In Arjuna, this definition corresponds to: in-flight transactions are those BasicActions that have created and not yet begun, or a status() of ActionStatus.RUNNING

Reply:
I'll copy @mmusgrov on my contribution here as I think that the change that @jmfinelli pushed to address an earlier comment on how to define in-flight means the discussion is no longer showing here in the diff view (but can be seen https://github.com/jbosstm/narayana/pull/2277#discussion_r1692879991) and my comment is also about definition.

That being said, the wording that @jmfinelli has used is broadly in line with what I said but I still think it's worth to define it from BasicAction point of view (and I am still wondering if we should maybe be counting CREATED status).

Reply:
The minimum status is [ActionStatus.RUNNING](https://github.com/jbosstm/narayana/blob/main/ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/ActionStatus.java#L24).

Reply:
> "Note: The definition of in-flight transactions are those BasicActions that have status() of ActionStatus.RUNNING"

Manuel is [already using that definition](https://github.com/jbosstm/narayana/pull/2277/files#diff-e06a851e6fc1d3ec189ce270be29f80e4e323ffbbc27795fbc9c8cb32e759da6R60).

Reply:
To include the ABORT_ONLY nuance that Michael suggested, we can go for something like:

Note: The definition of in-flight (AKA active) transactions, can be found in the OMG OTS specification: [active is defined as] the state of a transaction when processing is in progress and completion of the transaction has not yet commenced. In addition to the OMG OTS' definition, transactions that are marked as abort only are not considered to be active as they have not started the 2PC protocol. In Arjuna, these consideration boiled down to: in-flight transactions are those BasicActions that have status() equal to ActionStatus.RUNNING.

Reply:
And what Manuel is doing in the PR is [consistent with the JTA mapping](https://github.com/jbosstm/narayana/blob/main/ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/utils/arjunacore/StatusConverter.java#L15).

Reply:
> "Note: The definition of in-flight transactions are those BasicActions that have status() of ActionStatus.RUNNING"

But if we want to side step the nuances of using the JTA definition (which I believe is consistent with the ArjunaCore definition) then we could just go with this and be done.

Reply:
I am OK with either definition, I'll resolve this therefore. I am going to make the point about CREATED where the filtering is.

Reply:
I'll use this javadoc then:

Note: The definition of in-flight (AKA active) transactions, can be found in the OMG OTS specification: [active is defined as] the state of a transaction when processing is in progress and completion of the transaction has not yet commenced. In addition to the OMG OTS' definition, transactions that are marked as abort only are not considered to be active as they have not started the 2PC protocol. In Arjuna, these consideration boiled down to: in-flight transactions are those BasicActions that have status() equal to ActionStatus.RUNNING.